cylc/flow/scheduler.py

```diff
@@ -123,16 +124,55 @@ class SchedulerError(CylcError):
     pass
 
 
+class Timer:
+    """Simple timer class to use for various workflow timers.
+
+    """
+    def __init__(self, name, interval, log_reset_func=None):
+        """Initialize a timer."""
+        if log_reset_func is not None:
+            self.log_timer_reset = log_reset_func
+        else:
+            self.log_timer_reset = LOG.warning
+        self.name = re.sub('timeout', 'timer', name)
+        self.interval = get_interval_str(interval)
+        self.interval_float = interval
+        self.timeout = None
+
+    def reset(self):
+        """Start the timer now (by setting a concrete timeout value)."""
+        self.timeout = time() + self.interval_float
+        self.log_timer_reset(f"{self.interval} {self.name} timer starts NOW")
```

Comment:
With the `re.sub()` above, "timer" is already in `self.name`
```suggestion
        self.log_timer_reset(f"{self.interval} {self.name} starts NOW")
```